packages/react-badge/src/components/CounterBadge/useCounterBadgeStyles.ts

```diff
@@ -37,7 +38,7 @@ const useStyles = makeStyles({
 /**
  * Applies style classnames to slots
  */
-export const useCounterBadgeStyles = (state: CounterBadgeState) => {
+export const useCounterBadgeStyles: (state: CounterBadgeState) => BadgeState = (state: CounterBadgeState) => {
```

Comment:
This should should return `CounterBadgeState`

```suggestion
export const useCounterBadgeStyles = (state: CounterBadgeState): CounterBadgeState => {
```

➕

```diff
    !state.showZero && state.count === 0 && !state.dot && styles.hide,
    state.className,
  );
- return useBadgeStyles(state);
+ return useBadgeStyles(state) as CounterBadgeState;
```